### PR TITLE
fix(systemtrayicon): #3419, don't activate qTox widget on tray icon click in Unity backend

### PR DESCRIPTION
**Pull request by sudden6**
based on #3423 by @abbat

---

This change is [Reviewable](https://reviewable.io/reviews/tux3/qtox/3580)

### CODE REVIEW

File: src/widget/widget.cpp [hunk 1]
```diff
@@ -1818,7 +1818,13 @@ void Widget::onTryCreateTrayIcon()
             trayMenu->addAction(actionQuit);
             icon->setContextMenu(trayMenu);
 
-            connect(icon, &SystemTrayIcon::activated, this, &Widget::onIconClick);
+            // don't activate qTox widget on tray icon click in Unity backend (see #3419)
+#ifdef ENABLE_SYSTRAY_UNITY_BACKEND
```
Comment (antis81) [inline src/widget/widget.cpp]: ifdef not needed here, why not use the backend query?

Comment (sudden6) [inline src/widget/widget.cpp]: probably an optimization attempt, but not needed here.

Comment (antis81) [inline src/widget/widget.cpp]: ??

Comment (antis81) [inline src/widget/widget.cpp]: Please remove that ifdef. It complicates release management for linux by adding requirement to specifically compile for unity and "everything else".

Comment (sudden6) [inline src/widget/widget.cpp]: an attempt to optimize away the "if"^^ As you correctly saw it was useless.

**Comment by antis81** [comment]
:lgtm: But I'm not using unity. @sudden6 I suppose you do. If you confirm, it's ready to merge.

**Comment by sudden6** [comment]
@antis81 I don't use it, but @abbat is using a similiar patch in the OBS builds, and so far nobody complained :)

**Comment by zetok** [comment]
Hi,

As you might be aware, repo has been moved under `qTox` org on github.

What does it have to do with your PR, you might ask.

Well, it doesn't change all that much, except that workflow has been slightly improved – now there are required checks that PR needs to pass in order to be merged.

Those checks are:
- travis (checks whether PR builds & commit message format)
- reviewable (requirement for PR to be reviewed & accepted)

Since osx travis was broken for a bit, you will need to rebase your PR on newest master to get it to pass travis check. Sorry for inconvenience.

Reminder: as always, PR needs to be possible to merge without conflicts. Make sure it does ;)

**Comment by agilob** [comment]
:+1:

**Comment by sudden6** [comment]
@agilob if :+1:   equals LGTM could you please do it on reviewable? else we won't be able to merge.

**Comment by sudden6** [comment]
@antis81 now I remember why the `#ifdef` was there, the `SystrayBackendType` enum doesn't contain `Unity` by default, so it won't compile.

Question is, why should the enum not contain all backends every time?

**Comment by zetok** [comment]
> Question is, why should the enum not contain all backends every time?

Why would it?
https://github.com/qTox/qTox/wiki/Tray-icon

**Comment by sudden6** [comment]
Because it would make no difference, since this enum isn't used to determine the active or available backends.

**Comment by antis81** [comment]
Reviewed 2 of 3 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3580)_

**Comment by antis81** [comment]
:lgtm:

---

Review status: all files reviewed at latest revision, all discussions resolved.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/qtox/qtox/3580#-:-KOdU66BE3n6DYZCDnLn:bnfp4nl)_

**Pull request merged**